app/services/cohort_service.rb

```diff
@@ -0,0 +1,39 @@
+class CohortService
+  attr_reader :region, :quarters
+
+  def initialize(region:, quarters: nil)
+    @region = region
+    @quarters = quarters || default_quarters
+  end
+
+  # Each quarter cohort is made up of patients registered in the previous quarter
+  # who has had a follow up visit in the current quarter.
+  def totals
+    result = {quarterly_registrations: []}
+    quarters.each do |results_quarter|
+      cohort_quarter = results_quarter.previous_quarter
+
+      period = {cohort_period: :quarter,
+                registration_quarter: cohort_quarter.number,
+                registration_year: cohort_quarter.year}
+      query = MyFacilities::BloodPressureControlQuery.new(facilities: region.facilities, cohort_period: period)
+      result[:quarterly_registrations] << {
+        results_in: format_quarter(results_quarter),
+        patients_registered: format_quarter(cohort_quarter),
+        registered: query.cohort_registrations.count,
```

Comment:
[Note to self]:`cohort_registrations` is being removed in the assigned facility migration.

Reply:
The replacement `cohort_patients` is what we wanna use here anyway 👍 all good.

Reply:
Ah right. I believe the daily and monthly registration numbers already come via `Facility.registered_hypertension_patients`.👍